client/src/app/pages/archetypes/components/archetype-form/archetype-form.tsx

```diff
@@ -0,0 +1,352 @@
+import React, { useMemo } from "react";
+import { useTranslation } from "react-i18next";
+import { AxiosError } from "axios";
+import { Control, useForm, Path } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import * as yup from "yup";
+
+import {
+  ActionGroup,
+  Button,
+  ButtonVariant,
+  Form,
+} from "@patternfly/react-core";
+
+import type { Archetype, Tag } from "@app/api/models";
+import {
+  HookFormPFGroupController,
+  HookFormPFTextArea,
+  HookFormPFTextInput,
+} from "@app/components/HookFormPFFields";
+import { Autocomplete } from "@app/components/Autocomplete";
+import { NotificationsContext } from "@app/components/NotificationsContext";
+import {
+  useFetchArchetypes,
+  useFetchArchetypeById,
+  useCreateArchetypeMutation,
+  useUpdateArchetypeMutation,
+} from "@app/queries/archetypes";
+import { duplicateNameCheck, getAxiosErrorMessage } from "@app/utils/utils";
+import { useFetchTagCategories } from "@app/queries/tags";
+
+interface ArchetypeFormValues {
+  name: string;
+  description?: string;
+  comments?: string;
+  criteriaTags: string[]; // TODO: string[] only works if tags are uniquely named globally
```

Comment:
I was wrong about this field needing to be surface on UI level. This is an internal hub construct. We only need to pass tags as they are passed on applications today.

Reply:
Humm -- I still think that the 2 tag files are needed on UI...
 1. Criteria, so people can choose the tags that go in to the archetype-application matching algo
 2. "Static" archetype, so people can choose tags that should be associated with applications just by being matched to the archetype

There is a difference between what types the form values have and what is in the data model.  Currently, the `Autocomplete` component only supports a single list of strings for everything.  That doesn't line up nicely with an array of Tags or similar.  Thankfully when I tested creating Tags from curl, the names did need to be globally unique (regardless of what tag category tags belong to), so the current impl of Autocomplete is sufficient.

Reply:
@sjd78 I think the hub is taking care of that logic for us. It should work similarly to how the application form works. Can just pass tags as they are. The hub should know which tag belongs to what category & what tags are used in the matching logic. 
![Screenshot 2023-09-11 at 12 21 30 PM](https://github.com/konveyor/tackle2-ui/assets/11218376/7b238a90-9f58-4c81-bbfc-9d1f7fe1650b)


As far as the autocomplete component goes, I think we should just show the list of tags with no category for this first pass. And address the missing visual component of showing the associated category after feature freeze.

Reply:
Sure.  The form will display whatever data is provided for an edit and just empty values for a new archetype.

The form interface uses a `string[]` instead of a `TagRef[]` so the `Autocomplete` component is easier to use (via the `ItemsSelect` wrapper component).

As for the "criteria tags" and "archetype tags" fields, they are both still required inputs on the enhancement: https://github.com/konveyor/enhancements/blob/90b827b68cc367284a66bf66f087d5c263487e05/enhancements/assessment-module/README.md#creating-new-archetypes

Are you talking about swapping the actual Archetype API model from `Tag[]` to `TagRef[]`?